Experimenting with Docker

jhannah opened the pull request:
WIP. Creating PR just to have something to easily show Chris Short, in case this is useful.

jhannah committed:
Experimenting with Docker

jhannah committed:
1 second was a lie. Usually COPY . . takes a second

Because, normally, at least 1 file in the repo will have changed.

jhannah committed:
Rough in deps, with lots of notes.

jhannah committed:
more clear note

jhannah committed:
hmm... well we can Docker COPY this file...

jhannah committed:
Added ENV per Slack, still can't get this thing to run

jhannah committed:
Comment out keycloak for now

jhannah committed:
nx run client:build (broken)

jhannah committed:
little notes

devcshort commented on Dockerfile:
```diff
@@ -0,0 +1,69 @@
+# RUN apk add keycloak
+# keytool      ... there's no way to see the version?
+
+COPY packages/client/next.config.js packages/client/.norse/next.config.js
```
This line and the one following should be removed. These get generated at run/build time

jhannah replied:
done, thanks

devcshort commented on Dockerfile:
```diff
@@ -0,0 +1,69 @@
+
+# Do all of these first so Docker can cache the result for faster DEV cycles
+# (we can "build" (from cache) in 2 seconds instead of 5 minutes)
+COPY package.json package.json
```
package-lock.json should also be copied as it contains the metadata needed to make sure the correct versions are installed of all packages

jhannah replied:
done, thanks

jhannah committed:
PR feedback

jhannah committed:
zomg! It's ~working!!!

jhannah committed:
ooo starts itself now :)

jhannah committed:
Elasticsearch: Works if I do it manually

but fails under Dockerfile -- docker is mad at the noise Alpine
Linux is throwing for reasons I can't figure out.

jhannah committed:
Attempting Docker Compose

jhannah committed:
woot! Server connecting to Elasticsearch

jhannah committed:
Redis working? (not exploding on startup)

jhannah committed:
Client and server both working?

jhannah committed:
pre-compose notes removed

jhannah committed:
Initialize Elasticsearch index

jhannah committed:
New init of 0-taxonomies_v2_en

devcshort committed:
Make suggestion list word wrap

devcshort committed:
Add ability to hide categories and last assured date

devcshort committed:
Add a config to hide categories and last assured date

jhannah committed:
attempting debugging

jhannah commented:
Crazy. I rebased and force pushed too many times and github got confused about what files are modified in this PR. So closing this so I can open a new one (where github works fine). See #2 instead.
